include/acl_thread.h

```diff
@@ -23,10 +27,17 @@ extern "C" {
 #define ACL_TLS __declspec(thread)
 #endif
 
+// Foward declaration
+class aclMutex;
```

Comment:
Sure, but we already have acl_mutex_t defined in acl_support, should that be a problem? It might be a little bit confusing but I guess other parts of the runtime does something similar like this.

Reply:
I agree with Peter. We should keep the `snake_case` and create a suitable name for your wrapper class. Ideally we can know some of the differences by reading the name and all of the differences by reading the comments of the class.